crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -420,3 +422,33 @@ impl RegisterDiagnostic for App {
         self
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_clear_history() {
+        let mut diagnostic =
+            Diagnostic::new(DiagnosticPath::new("test")).with_max_history_length(5);
+        let mut now = Instant::now();
+
+        // Three times in a row...
+        for _ in 0..3 {
+            // Fill the diagnostic with measurements...
+            for _ in 0..5 {
+                diagnostic.add_measurement(DiagnosticMeasurement {
+                    time: now,
+                    value: 20.0,
+                });
+                // Increase time to test smoothed average.
+                now += Duration::from_secs(1);
+            }
+            // Assert that the average and smoothed average are correct...
+            assert!((diagnostic.average().unwrap() - 20.0) < 0.1);
+            assert!((diagnostic.smoothed().unwrap() - 20.0) < 0.1);
```

Comment:
`(diagnostic.average().unwrap() - value).abs() < 0.1` would be slightly more robust